073-decouple-ca-management-from-end-entity-certificate-management.md

```diff
@@ -0,0 +1,339 @@
+<!-- This template is provided as an example with sections you may wish to comment on with respect to your proposal. Add or remove sections as required to best articulate the proposal. -->
+
+# Decouple CA certificate management from end-entity certificate management
+
+This proposal aims to decouple the management of the cluster CA certificate from the rolling out of trust of the CA certificate to components in a Strimzi cluster.
+
+## Current situation
+
+To provide a secure, TLS-enabled setup by default when deploying Kafka clusters, Strimzi integrated its own CA operations into the Cluster Operator.
+The Cluster Operator accomplishes this by using `openssl` to generate self-signed root CA certificates and private keys which it then uses to directly sign end-entity (EE) certificates (i.e. intermediate certificates are not used).
```

Comment:
```suggestion
The Cluster Operator accomplishes this by using `openssl` to generate self-signed root CA certificates and private keys which it then uses to directly sign end-entity (EE) certificates (i.e. intermediate certificates can't be used).
```

Not sure we want to add this level of details to the proposal, but the way to enforce what you describe here is to generate a self-signed CA with pathlen=0, which is precisely what Strimzi does.

Example:

```sh
Certificate:
    Data:
        ...
        Issuer: O=io.strimzi, CN=cluster-ca v0
        ...
        Subject: O=io.strimzi, CN=cluster-ca v0
        ...
        X509v3 extensions:
            ...
            X509v3 Basic Constraints: critical
                CA:TRUE, pathlen:0
            X509v3 Key Usage: critical
                Certificate Sign, CRL Sign
```